### PR TITLE
feat(tier4_planning_msgs): improve avoidance target filter

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Related Links

- https://github.com/autowarefoundation/autoware.universe/pull/2329

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Code is properly formatted
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code is properly formatted
- [x] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [x] All open points are addressed and tracked via issues or tickets
- [x] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR**: Required to pass before the merge.
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Comment by satoshi-ota** [comment]
@zulfaqar-azmi-t4 @TakaHoribe @rej55 
The related [PR](https://github.com/autowarefoundation/autoware.universe/pull/2329)'s CI/CD is blocked by this PR, so could you please approve this PR if there is no problem?

**Review by rej55 — APPROVED** [review]
LGTM

**Pull request merged**
